serviceLibrary/src/main/java/info/mqtt/android/service/MqttAndroidClient.kt

```diff
@@ -238,9 +238,9 @@ class MqttAndroidClient @JvmOverloads constructor(
     private fun registerReceiver(receiver: BroadcastReceiver) {
         val filter = IntentFilter()
         filter.addAction(MqttServiceConstants.CALLBACK_TO_ACTIVITY)
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-            context.registerReceiver(receiver, filter, Context.RECEIVER_NOT_EXPORTED)
-        } else
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU)
+            context.registerReceiver(receiver, filter, Context.RECEIVER_EXPORTED)
```

Comment:
Why must it be exported ?

Reply:
Not really sure about what I'm going to say but I think because the instrumentation test app and the library are on different places. It took me while to find out that the broadcasts send by the `callbackToActivity` function are not received by the `MqttAndroidClient` broadcast receiver. It fixed by this change.
It might work in real app

Reply:
Can we use this?
https://stackoverflow.com/questions/76919130/android-14-context-registered-broadcast-receivers-not-working

Reply:
I'm open for any improvment

Reply:
I will see what can I do